hecuba_core/src/SpaceFillingCurve.h

```diff
@@ -61,14 +68,16 @@ class SpaceFillingCurve {
 
         virtual int32_t computeNextClusterId() = 0;
 
-        virtual void *merge_partitions(const ArrayMetadata *metas, std::vector<Partition> chunks) = 0;
+        virtual void merge_partitions(const ArrayMetadata *metas, std::vector<Partition> chunks, void *data) = 0;
 
     };
 
 
     ~SpaceFillingCurve() {};
 
-    static PartitionGenerator *make_partitions_generator(const ArrayMetadata *metas, void *data);
+    static PartitionGenerator *
+    make_partitions_generator(const ArrayMetadata *metas, void *data,
```

Comment:
add method make_filtered_partitions_gen
When the partitioning algorithm does not support, raise ModuleException()